### ISSUE TITLE
Force parentheses for `a or b <op> c`

### ISSUE THREAD

**Issue by jonashaag** [astral-sh/ruff#19565]
### Summary

Example: `a or b > c`

What might be meant here: `(a or b) > c`

Actual precedence: `a or (b > c)`

Misleading example:

```py
amount: int | None

amount or 0 < 100

# with amount = 101 evaluates to 101, not False!
```

**Comment by gorewilliams** [comment]
I guess it actually makes sense that it doesn't currently support this as it probably treats expressions as something between `and`/`or` operators, where here it is a mathematical operator (`<`).

**Comment by gorewilliams** [comment]
Looks to me like something that should be added to `RUF021`

**Comment by MichaReiser** [comment]
RUF021 certainly has overlap but we've also seen in the past that users like to have very fine grain control. That might justify a separate rule. (for context, this is where RUF021 was introduced https://github.com/astral-sh/ruff/issues/8721)
